scripts/new_project.kts

```diff
@@ -41,6 +45,57 @@ object NewProject {
         }
     }
 
+    private fun renamePackageNameFolders() {
+        showMessage("=> 🔎 Rename the package folders...")
+        modules.forEach { module ->
+            val srcPath = projectPath + File.separator + module + File.separator + "src"
+            File(srcPath)
+                .walk()
+                .maxDepth(2)
+                .filter { it.isDirectory && it.name == "java" }
+                .forEach { javaDirectory ->
+                    val javaPath = javaDirectory.absolutePath
+                    val currentPackagePath =
+                        javaPath + File.separator + TEMPLATE_PACKAGE_NAME.replace(
+                            DOT_SEPARATOR,
+                            File.separator
+                        )
+                    val newPackagePath =
+                        javaPath + File.separator + packageName.replace(
+                            DOT_SEPARATOR,
+                            File.separator
+                        )
+                    if (File(currentPackagePath).exists()) {
+                        copyCurrentPackageToNewPackageAndClean(
+                            currentPackagePath = currentPackagePath,
+                            tempPath = javaPath + File.separator + "temp_folder",
+                            newPackagePath = newPackagePath
+                        )
+                    }
```

Comment:
This solution works well, but I was hoping we could make use of `renameTo()` so we don't require `temp_folder`.

What do you think about this solution? 🙇 

If it works we can minimize the code 🔥 

```suggestion
                   val oldDirectory = File(
                        javaDirectory, TEMPLATE_PACKAGE_NAME.replace(
                            oldValue = DOT_SEPARATOR,
                            newValue = File.separator
                        )
                    )
                    val newDirectory = File(
                        javaDirectory, packageName.replace(
                            oldValue = DOT_SEPARATOR,
                            newValue = File.separator
                        )
                    )
                    newDirectory.mkdirs()
                    oldDirectory.renameTo(newDirectory)
                    oldDirectory.parentFile.parentFile.deleteRecursively()
```

Reply:
@Tuubz this solution looks good, but it's failed in case of the new package path is similar to the old package path, ex `co.nimblehq.testproject` 🙏   The new package folder will be deleted 😓 
Here's the result when I run the above solution 😂  `java` folder is empty.
![image](https://user-images.githubusercontent.com/60863885/169239903-849c0164-e76d-4627-beae-fd7fc147b516.png)

That's why I create a `temp` folder to store all files and then delete the old package folder.
Do we have other solutions for this case? 👉  Yes, it requires the calculation to delete the correct folder (delete only `testproject` in this case) but I think we can go with my simple solution (use `temp` folder to store files temporarily) instead.
If you have other ideas, please let me know 😁  I'm open to discussing more.

Reply:
Good catch, I missed that one! 👍 

OK, I agree that the temp folder is necessary to make this work

However, I do have an urge to simplify it a bit, for easier future maintainability 🙇 

What do you think about this solution? (It also makes use of `temp_folder`):

```
File(srcPath)
        .walk()
        .maxDepth(2)
        .filter { it.isDirectory && it.name == "java" }
        .forEach { javaDirectory ->
            val oldDirectory = File(
                javaDirectory, TEMPLATE_PACKAGE_NAME.replace(
                    oldValue = DOT_SEPARATOR,
                    newValue = File.separator
                )
            )
            val newDirectory = File(
                javaDirectory, packageName.replace(
                    oldValue = DOT_SEPARATOR,
                    newValue = File.separator
                )
            )

            val tempDirectory = File(javaDirectory, "temp_directory")
            oldDirectory.copyRecursively(tempDirectory)
            oldDirectory.parentFile.parentFile.deleteRecursively()
            newDirectory.mkdirs()
            tempDirectory.copyRecursively(newDirectory)
            tempDirectory.deleteRecursively()
        }
```

Last suggestion ❤️

Reply:
> @Tuubz this solution looks good, but it's failed in case of the new package path is similar to the old package path, ex `co.nimblehq.testproject` 🙏 The new package folder will be deleted 😓 Here's the result when I run the above solution 😂 `java` folder is empty. ![image](https://user-images.githubusercontent.com/60863885/169239903-849c0164-e76d-4627-beae-fd7fc147b516.png)
> 
> That's why I create a `temp` folder to store all files and then delete the old package folder. Do we have other solutions for this case? 👉 Yes, it requires the calculation to delete the correct folder (delete only `testproject` in this case) but I think we can go with my simple solution (use `temp` folder to store files temporarily) instead. If you have other ideas, please let me know 😁 I'm open to discussing more.

The folder is deleted because of the thread I think. The script got executed line by line and the old folder got deleted while it is renaming the new folder.